libs/langchain/ragstack_langchain/colbert/embedding.py

```diff
@@ -0,0 +1,58 @@
+from typing import List, Optional
+
+from langchain_core.embeddings import Embeddings
+from ragstack_colbert import DEFAULT_COLBERT_MODEL, ColbertEmbeddingModel
+from ragstack_colbert.base_embedding_model import BaseEmbeddingModel
+from typing_extensions import override
+
+
+class TokensEmbeddings(Embeddings):
+    """Adapter for token-based embedding models and the LangChain Embeddings."""
+
+    def __init__(self, embedding: BaseEmbeddingModel = None):
+        self.embedding = embedding or ColbertEmbeddingModel()
+
+    @override
+    def embed_documents(self, texts: List[str]) -> List[List[float]]:
```

Comment:
This doesn't seem right. It changes the signatures of the `Embeddings` class in an incompatible way. I think if we want to go this direction, we should instead have:

1. `TokensEmbeddings` is an interface/ABC like this, but does not extend `Embeddings`.
2. We have a sub-class of `Embeddings` that defines `get_tokens_embeddings() -> TokensEmbeddings` (or just rely on the method being defined).
3. So then subclasses are valid langchain embeddings, and can choose to implement `get_tokens_embeddings()` to get the different interface.

Reply:
I don't get it. In its current form `TokensEmbeddings` is a valid `Embeddings` subclass.
I'm not sure there is a lot of value in introducing yet another interface.
Here `TokensEmbeddings` is a simple wrapper around `BaseEmbeddingModel` that implements `Embeddings` to please the `VectorStore` API.

Reply:
Ah, yeah. I misread this. It does match the interface.